test/testMPAppIDs.cpp

```diff
@@ -0,0 +1,37 @@
+#ifndef TESTMPAPPIDS_H
+#define TESTMPAPPIDS_H
+
+#include "pmpo_createTestMPMesh.hpp"
+#include "pmpo_defines.h"
+#include "pmpo_c.h"
+
+extern "C" {
+  void testAppIDPointer(MPMesh_ptr p_mpmesh);
+}
+
+void testAppIDPointer(MPMesh_ptr p_mpmesh) {
+  auto p_MPs = ((polyMPO::MPMesh*)p_mpmesh)->p_MPs;
+
+  int numAddedMPs = 2;
+  Kokkos::View<int*> added_mp2Elm_d("added_mp2Elm_d", numAddedMPs);
+  Kokkos::parallel_for("set addedMP2Elm", numAddedMPs, KOKKOS_LAMBDA (const int i) {
+    added_mp2Elm_d(i) = i;
+  });
+
+  std::vector<int> added_mpIDs(numAddedMPs);
+  for(int i=0; i<numAddedMPs; i++)
+    added_mpIDs[i] = p_MPs->getNextAppID();
+  auto added_mpIDs_d = create_mirror_view_and_copy(added_mpIDs.data(), numAddedMPs);
+  int prevNumMPs = p_MPs->getCount();
+
+  p_MPs->rebuild(added_mp2Elm_d, added_mpIDs_d);
+
+  //Assert rebuild worked
+  auto newAppID = p_MPs->getData<polyMPO::MPF_MP_APP_ID>();
+  Kokkos::parallel_for("print APP ID", numAddedMPs, KOKKOS_LAMBDA (const int i) {
+    printf(" Added %d\n", newAppID(prevNumMPs+i));
+    assert(added_mpIDs_d[i] == newAppID(prevNumMPs+i));
```

Comment:
I feel this test is a little fragile, e.g., what if MP order changes within PUMIpic during rebuild (when migrated-in MPs are added).
One option: may be we can get APP ID for all MPs and then make sure each ID exists once only and make sure 'added_mpIDs_d' are also present.
Second option: may be we use ELM ID of MPs (added_mp2Elm_d and assuming each added MP has a local index which is its ELM ID too , from above: `added_mp2Elm_d(i) = i;`), and then assert on that ELM ID while using MP APP ID, e.g., pseudo code as follows: mpElmID = getMPElmID() and then mpElmID[mpAppID[i]] = added_mp2Elm_d[i].

Reply:
I think I need to re-think on the second option that I suggested. I think it should be something like:
mpAppID = getMPAppID() and mpElmID = getMPElmID() (both on device) and then copy mpAppID and mpElmID to host where sizing mpAppID_h and mpElmID_h arrays will require prevNumMPs+numAddedMPs, and then within a serial-for with iterator i_mp and check for each entry in mpElmID_h (for each i_mp index) matches with which j-th entry in added_mp2Elm_d and then assert mpAppID_h[i_mp]==j. Looks complicated and not fully sure on this.

Reply:
I agree that the test is fragile, but we could leave it since the test is so simple and self contained. Otherwise I could use what we do for our pumi-pic tests. Which is that we send our local index with the rebuild and then we retrieve the local indexes to check that it got set to the correct element.

Reply:
OK, in this case: we can remove the test or I suggest to then use a std::map in the C++ test.
My preference is for adding std::map based test, something relatively simple like:
std::map <int mp_app_ID, int elm_ID>  existing_MPs_data;
... fill existing_MPs_data

std::map <int mp_app_ID, int elm_ID>  added_MPs_data;
... fill added_MPs_data

after rebuild

get MP data using getMPAppIDs and getElmIDs for all MPs (existing and added)
loop over each MP App ID and make sure those that are not in existing_MPs_data are in added_MPs_data
   also assert on elmID for each added MP and change added_MPs_data_iter.second = MP_ELM_INVALID
   countAddedMPs
end loop
assert on countAddedMPs

Let me know if any questions.